### PR TITLE
add modf to PMacc::algorithms::math

### PR DESCRIPTION
**Pull request by Heikman**
add modf to PMacc::algorithms::math

### CODE REVIEW

File: src/libPMacc/include/algorithms/math/defines/modf.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{
+
+namespace math
+{
+
+template<typename Type>
+struct Modf;
+
+template<typename T>
+HDINLINE typename Modf< T>::result modf(T value, T* intpart)
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/defines/modf.hpp]: ` Modf< T>` is one space missing or to much.

File: src/libPMacc/include/algorithms/math/doubleMath/modf.tpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+template<>
+struct Modf<double>
+{
+    typedef double result;
+
+    HDINLINE double operator( )(double value, double* intpart)
+    {
+#if __CUDA_ARCH__
+        return ::modf( value, intpart);
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/doubleMath/modf.tpp]: `::modf( value, intpart);` one space to much else it not look like line 46

File: src/libPMacc/include/algorithms/math/floatMath/modf.tpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+template<>
+struct Modf<float>
+{
+    typedef float result;
+
+    HDINLINE float operator( )(float value, float* intpart)
+    {
+#if __CUDA_ARCH__
+        return ::modf( value, intpart );
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/floatMath/modf.tpp]: Please choose your space style :-) all of your implementation should be `function ( x, y );` or `function (x, y);`

File: src/libPMacc/include/algorithms/math/floatMath/modf.tpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+template<>
+struct Modf<float>
+{
+    typedef float result;
+
+    HDINLINE float operator()(float value, float* intpart)
+    {
+#if __CUDA_ARCH__
+        return ::modf(value, intpart);
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/floatMath/modf.tpp]: please use `::modff`

File: src/libPMacc/include/algorithms/math/floatMath/modf.tpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#else
+        return std::modf(value, intpart);
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/floatMath/modf.tpp]: please use `std::modff`

**Comment by psychocoderHPC** [comment]
please update [this](https://github.com/ComputationalRadiationPhysics/picongpu/pull/1218#discussion_r43890563)

**Comment by Heikman** [comment]
beep

**Comment by ax3l** [comment]
How does CUDA's `modf` and `std::modf` behave for negative values?
Do they behave the same?
Can you pls add a documentation string to the functor?

**Pull request merged**